packages/flutter_tools/test/general.shard/terminal_handler_test.dart

```diff
@@ -27,6 +28,7 @@ void main() {
             trackWidgetCreation: false,
             treeShakeIcons: false,
           ),
+          generator: MockResidentCompiler(),
```

Comment:
If the generator is not provided, one is created but that looks up the dartSdk path from the cache.

Reply:
🤦

Reply:
I thought it was hard to write tools tests, but turns out I had no idea...

Reply:
the nice thing is ... there are no issues with the tests using testWithoutContext, so at least we are on the right path 👍